src/ImageSharp/Common/Helpers/Vector256Utilities.cs

```diff
@@ -73,6 +82,34 @@ public static Vector256<byte> Shuffle(Vector256<byte> vector, Vector256<byte> in
         return default;
     }
 
+    /// <summary>
+    /// Performs a conversion from a 256-bit vector of 8 single-precision floating-point values to a 256-bit vector of 8 signed 32-bit integer values.
+    /// Rounding is equivalent to <see cref="MidpointRounding.ToEven"/>.
+    /// </summary>
+    /// <param name="vector">The value to convert.</param>
+    /// <returns>The <see cref="Vector256{Int32}"/>.</returns>
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    public static Vector256<int> ConvertToInt32RoundToEven(Vector256<float> vector)
```

Comment:
Personally, I'd avoid helpers like this for now.  As of .NET 8, only AVX hardware supports hardware-accelerated Vector256 at all, so this is just that with more steps.  You may be able to expand that when Arm SVE becomes available in .NET 9 (or 10 if you stick to the current pattern of targeting LTS), but one would hope by that time, you'll have a BCL-provided xplat helper you can use to replace any explicit AVX calls.

That is to say, at the time this becomes actually useful, you'll have to make changes anyway.

Same goes for Vector512, ofc.

Reply:
The hope is that bythe time 10 comes out I can replace `Vector256Utilities.XX` with `Vector256.XX` and it will be a simple job to refactor. Plus, the consistency feeds my soul.

Reply:
> the consistency feeds my soul

I feel this 😄

My concern would be that the fallback code here is either
1) Never used, so just a maintenance burden with no payoff, or
2) Used accidentally, resulting in a perf drop over the optimized `Vector128` path

Perhaps if you really do want to hide all intrinsics behind helpers like this, it would be better to throw PNSE outside the one optimized implementation.  Be aware, though, adding these extra layers over intrinsics can hinder inlining when they're used in larger methods.

When refactoring time comes, most of these will hopefully just be `Avx2.XX` -> `Vector256.XX` anyway.